File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(0, head)
        cur = head
        while n > 0 and cur:
            cur = cur.next
            n -= 1
        slow = dummy
        while cur:
            slow = slow.next
            cur = cur.next
        slow.next = slow.next.next
        return dummy.next

File: Data Structures & Algorithms/reverse-linked-list-ii/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        ListNode dummy = new ListNode(-1);
        dummy.next = head;
        ListNode cur = dummy;
        for(int i = 0; i < left - 1; i++) {
            cur = cur.next;
        }
        ListNode prevTail = cur;
        ListNode nextTail = prevTail.next;
        ListNode pre = null;
        cur = nextTail;
        for(int i = left; i <= right; i++) {
            ListNode nxt = cur.next;
            cur.next = pre;
            pre = cur;
            cur = nxt;
        }
        prevTail.next = pre;
        nextTail.next = cur;
        return dummy.next;
    }
}